nabs/callbacks.py

```diff
@@ -0,0 +1,86 @@
+"""
+Callbacks for subscription to the Bluesky ``RunEngine``
+
+This is the LCLS counterpart to `bluesky.callbacks`
+
+Callbacks in this module expect and operate on the `start`, `event`,
+`stop`, etc documents.
+"""
+
+import logging
+
+from bluesky.callbacks.core import CallbackBase, make_class_safe
+
+logger = logging.getLogger(__name__)
+
+
+@make_class_safe(logger=logger)
+class ELogPoster(CallbackBase):
+    """
+    Callback to post messages to the elog, primarily related to plan setup and
+    results. This callback relies on and must be instantiated after the
+    BestEffortCallback and HutchELog.
+
+    To subscribe:
+
+    .. code-block:: python
+
+        elogc = ELogPoster(bec, elog)
+        elogc_uid = RE.subscribe(elogc)
+
+    To enable posting for a specific run:
+
+    .. code-block:: python
+
+        RE(plan(), post=True)
+
+    Parameters
+    ----------
+    bec : BestEffortCallback
+        Instance of BestEffortCallback subscribed to the current RunEngine
+
+    elog : HutchELog
+        Instance of HutchELog connected to the desired logbook
+    """
+    def __init__(self, bec, elog):
+        self._bec = bec
+        self._elog = elog
+
+        # default to elog setting
+        self._send_post = elog.enable_run_posts
+
+        # TO-DO:
+        # - add granular switches as optional arguments
+        # - add error handling for misspelled keys
+        # - add default lists for start doc keys, elog tags?
+
+    def start(self, doc):
+        """ Post plan information on start document"""
+        if 'post' in doc.keys():
+            # Override default if key exists
+            self._send_post = doc['post']
+        else:
+            self._send_post = self._elog.enable_run_posts
+
+        if self._send_post:
+            run_info = str({k: doc[k] for k in ['plan_name', 'plan_args']})
```

Comment:
Should this be a `json.dumps`? What is the elog post expecting?

Reply:
I'll check with Murali, the only supported formats I saw were plain text and html.  we we can submit any arbitrary attachment, but I think we want this information to be easily readable

Reply:
There's a point to be made here that posts to the elog should be human-readable, so a json is maybe not a desirable format.  

There's the option to post to html, maybe we try to auto-format some of that.  Though the run-start data can become rather deeply nested.